pkg/git/service.go

```diff
@@ -188,6 +192,134 @@ func (s *Service) SetGitConfig(userData *gitprovider.GitUser) error {
 	return nil
 }
 
+func formatCommit(commit *object.Commit, branchName string) project.CommitInfo {
+	return project.CommitInfo{
+		Hash:    commit.Hash.String(),
+		Author:  commit.Author.Name,
+		Message: strings.TrimSpace(commit.Message),
+		Date:    commit.Author.When.Format("2006-01-02 15:04:05"),
+		Branch:  branchName,
+	}
+}
+
+func (s *Service) GetUnpushedCommitsInfo() ([]project.CommitInfo, error) {
```

Comment:
https://github.com/daytonaio/daytona/pull/1103#issuecomment-2360919228

Based on the comment above, this functions should use `Exec` to execute `git rev-list ....` and then parse the output. The function should return the 2 numbers that represent the number of commits different in each branch.